SharpXMPP.Shared/Resolver.cs

```diff
@@ -18,12 +20,15 @@ public struct SRVRecord
 
     public static class Resolver
     {
-        public async static Task<List<SRVRecord>> ResolveXMPPClient(string domain)
+        [Obsolete("Call overload with CancellationToken")]
+        public static Task<List<SRVRecord>> ResolveXMPPClient(string domain) =>
```

Comment:
This resolver should be replaced with DnsClient, will it be possible to do the same with DnsClient?

Reply:
From what I see, `DnsClient` (the one used in `SharpXMPP.XmppWebSocketConnection.ConnectAsync`, right?) seems to support cancellation as well. So, yeah, we'll be able to keep this anyway.

Reply:
Ah, would you like me to go ahead and rewrite this using `DnsClient`? I could, but let's do in in another PR.

Reply:
We have an issue for that - #71 - sure it is a different, no need to rewrite right now